R/audit-trail.R

```diff
@@ -0,0 +1,200 @@
+#' AuditLog Class
+#'
+#' This class is used internally to store audit logs for each request.
+AuditLog <- R6::R6Class( # nolint: object_name_linter.
+  "AuditLog",
+  public = list(
+    initialize = function(request_method, endpoint_url, request_body) {
+      private$request_id <- uuid::UUIDgenerate()
```

Comment:
In this case, request ID can be obtained from another sources in the future. For example, this might be a correlation ID from another system that uses unbiased and will set appropriate HTTP Header that will contain this ID. We can remove this for now if you don't think it will be useful, since the mechanism of capturing external request ID is not yet implemented.